enhancements/sig-architecture/81-addon-lifecycle/README.md

```diff
@@ -37,26 +38,787 @@ proposal's scope.
+Then the addon manager will generate `ManifestWork` with annotations `configsSpecHash`.
+
+Next, the addon manager updates the `LastAppliedConfigSpecHash` by checking the `ManifestWork`.
+  - `ManifestWork` annotations `configsSpecHash` matches `DesiredConfigSpecHash`.
+  - `ManifestWork` Condition Available status is true.
+  - `ManifestWork` Condition Available observedGeneration equals to generation.
+  - If it is a fresh install since one addon can have multiple ManifestWorks, the `ManifestWork` condition 
+  ManifestApplied must also be true.
+
+The addon manager then compares the `LastAppliedConfigSpecHash` with `DesiredConfigSpecHash` to update the condition.
+- If `LastAppliedConfigSpecHash` == `DesiredConfigSpecHash`
+  - If `LastAppliedConfigSpecHash` is empty before it equals to `DesiredConfigSpecHash`, update condition Progressing 
+  status to false with reason `InstallSucceed`.
+  - If `LastAppliedConfigSpecHash` is not empty before it equals to `DesiredConfigSpecHash`, update condition Progressing 
+  status to false with reason `UpgradeSucceed`.
+- If `LastAppliedConfigSpecHash` != `DesiredConfigSpecHash`
+  - If `LastAppliedConfigSpecHash` is empty, update condition Progressing status to true with reason `Installing`.
+    - If meets some error during the installing, for example, the `ManifestWork` reports an error, update condition Progressing 
+    status to false with reason `InstallFailed`.
+  - If `LastAppliedConfigSpecHash` is not empty, update condition Progressing status to true with reason `Upgrading`.
+    - If meets some error during the installing, for example, the `ManifestWork` reports an error, update condition Progressing 
+    status to false with reason `UpgradeFailed`.
+
+### Workflow
+Below is an example of `ClusterManagementAddOn`, `ManagedClusterAddOn` and `ManifestWork`:
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: AddOnHubConfig
+metadata:
+  name: hub-config-yyy
+spec:
+  desiredVersion: v0.10.0
+```
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddOn
+metadata:
+  name: helloworld
+spec:
+  configuration:
+    configs:
+    - group: addon.open-cluster-management.io
+      resource: addondeploymentconfigs
+      name: deploy-config-xxx
+      namespace: open-cluster-management
+    - group: addon.open-cluster-management.io
+      resource: addonhubconfigs
+      name: hub-config-yyy
+    rolloutStrategy:
+      type: RollingUpdateWithPlacement
+      rollingUpdateWithPlacement:
+        name: canary-placement
+        namespace: default
+        maxConcurrentlyUpdating: 25%
+  installStrategy:
+    type: Placements
+    placements:
+    - name: aws-placement
+      namespace: default
+      addonSpec:
+        installNamespace: open-cluster-management-agent-addon
+        configs:
+          - group: addon.open-cluster-management.io
+            resource: addondeploymentconfigs
+            name: aws-config
```

Comment:
Is someone going to ask if we can have a canary for changing this as well?  Would we do it by having an update strategy per `.spec.installStrategy.placement`?  That may be practical.  Perhaps my rolloutStrategy suggestion should have been under `.spec.installStrategy.placement`?

Can we try to set up a synchronous meeting to discuss this nesting aspect?

Reply:
Yes, a nested rollout strategy makes more sense, have updated the proposal with our latest ideas.